src/zoom.js

```diff
@@ -282,23 +282,14 @@ export default function() {
     for (i = 0; i < n; ++i) {
       t = touches[i], p = touch(this, touches, t.identifier);
       p = [p, this.__zoom.invert(p), t.identifier];
-      if (!g.touch0) g.touch0 = p, started = true;
-      else if (!g.touch1) g.touch1 = p;
+      if (!g.touch0) g.touch0 = p, started = true, g.tap = true, g.consecutive = !!touchstarting;
+      else if (!g.touch1) g.touch1 = p, g.tap = false;
```

Comment:
optimistically set "tap" field on gesture to true, then set to false if anything makes it not a tap.